enumerable.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  def my_each
+    return to_enum unless block_given?
+
+    length.times do |x|
+      yield(self[x])
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times do |x|
+      yield(x, self[x])
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    item = []
+    length.times do |x|
+      item.push(self[x]) if yield(self[x])
+    end
+    item
+  end
+
+  def my_all(field = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    return true if !block_given? && field.nil?
+
+    value = true
```

Comment:
- [ ] my_all? returns true if the block never returns false or nil
- [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil
- [ ] my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern